compiler/resolution/implementForallIntents.cpp

```diff
@@ -778,7 +778,7 @@ static void resolveShadowVarTypeIntent(LoopWithShadowVarsInterface* fs,
   }
 
   // Prune, as discussed in the above comment.
-  if (intent == TFI_REF)
+  if (intent == TFI_REF && fs->isForallStmt())
```

Comment:
What was the reason for this change? In any case, it may be a good idea to put a short note about it here.

Reply:
Changed to:

```C++
    // Prune, as discussed in the above comment.
    //
    // However, for foreach, we do not want to prune (replace a ref intent'd
    // shadow variable with its definition outside the loop) because if the loop
    // is gpuized we need to pass that variable by pointer. We do special
    // processing to handle this case in iterator lowering and gpu transforms and
    // as such do not want to prematurely prune it.
    if (intent == TFI_REF && fs->isForallStmt())
      prune = true;
```